core/arch/arm/plat-imx/pm/psci.c

```diff
@@ -55,31 +54,32 @@ int psci_cpu_on(uint32_t core_idx, uint32_t entry,
 	generic_boot_set_core_ns_entry(core_idx, entry, context_id);
 
 	val = virt_to_phys((void *)TEE_TEXT_VA_START);
+#ifdef CFG_MX7
```

Comment:
do you still need the test below?

Reply:
yes currently i still need that otherwise code will not compile.

Reply:
?
This may not be a good reason.

I was talking about `if (soc_is_imx7ds()) {`. If there are MX7 Socs without SRC support, i makes sense. (but i believe the function should return "not supported" instead of "success", but i gues such cores will never call psci services).

Reply:
there are multiple issue with out the i.MX plat is build, and this patch is not meant to adress all the issue.
out prefered way in th elong run is to have the soc_is_imx** type of function and less rely on CFG_MXxx in common code.
unfortunatly we are not quite there.
This is why we kept soc_is_imx7ds but if that makes the review goes smoother i will remove the check and will reintroduce that once the CFG_MX7 is gone

Reply:
Yes the long term plan is still be able to get one binary (or at least one binary for the CA9 one for CA7 and armv8), while keeping the possibility to compile individual SOC.
But clearly today the code is not in the shape to do so. and even worse than that there is a platform flavor for each and every board
so imho my claim is still valid enventyhough i have added some CFG_MX7, i have not broken any functionality
so if you don't mind i would like to keep the patch as is for now.

Btw eventhough PIC code /non flat mapping will be nice (we have started to look into non-flat mapping) if you want to reduce the number of binaries today , there is a simple solution. 
instead of using the end of the DDR for op-tee , use the begining of the memory, and  DT to detect the UART to use.
with those 2 simple modifications we should be able to have on common binary for all the i.MX6 D/Q/DL on  sabreauto and sabresd, and another binary for 6ul/7d.